src/Console/Command/SelfUpdateCommand.php

```diff
@@ -0,0 +1,196 @@
+            )
+        ;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->logger = new UpdateConsoleLogger(
+            new SymfonyStyle($input, $output)
+        );
+
+        $this->output = $output;
+
+        $this->version = $this->getApplication()->getVersion();
+
+        if ($input->getOption('rollback')) {
+            $this->rollback();
+
+            return 0;
+        }
+
+        if ($input->getOption('check')) {
+            $this->printAvailableUpdates();
+
+            return 0;
+        }
+
+        $this->update($this->createUpdater());
+    }
+
+    private function createUpdater(): Updater
+    {
+        $this->updater->setStrategy(Updater::STRATEGY_GITHUB);
+        $this->updater->getStrategy()->setPackageName(self::PACKAGIST_PACKAGE_NAME);
+        $this->updater->getStrategy()->setPharName(self::REMOTE_FILENAME);
+        $this->updater->getStrategy()->setCurrentLocalVersion($this->version);
+
+        return $this->updater;
+    }
+
+    private function update(Updater $updater)
+    {
+        $this->logger->startUpdating();
+        try {
+            $result = $this->updater->update();
+
+            $newVersion = $this->updater->getNewVersion();
+            $oldVersion = $this->updater->getOldVersion();
+
+            if ($result) {
+                $this->logger->updateSuccess($newVersion, $oldVersion);
+            } else {
+                $this->logger->updateNotNeeded($oldVersion);
+            }
+        } catch (\Throwable $e) {
+            $this->logger->error($e);
+        }
+    }
+
+    private function rollback()
+    {
+        try {
+            $result = $this->updater->rollback();
+            if ($result) {
+                $this->logger->rollbackSuccess();
+            } else {
+                $this->logger->rollbackFail();
+            }
+        } catch (\Throwable $e) {
+            $this->logger->error($e);
+        }
+    }
+
+    private function printAvailableUpdates()
+    {
+        $this->logger->printLocalVersion($this->version);
+        $this->printCurrentStableVersion();
+    }
+
+    private function printCurrentStableVersion()
+    {
+        $updater = $this->createUpdater();
+        $stability = self::STABILITY_STABLE;
+
+        try {
+            if ($updater->hasUpdate()) {
+                $this->logger->printRemoteVersion(
+                    $stability,
+                    $updater->getNewVersion()
+                );
+            } elseif (false == $updater->getNewVersion()) {
+                $this->logger->noNewRemoteVersions($stability);
+            } else {
+                $this->logger->currentVersionInstalled($stability);
+            }
+        } catch (\Throwable $e) {
+            $this->logger->error($e);
+        }
+    }
+
+    private function getLocalPharName(): string
+    {
+        return basename(\PHAR::running());
```

Comment:
Can you clarify? The SelfUpdateCommand only ever exists in a PHAR, so the above `running()` call should always succeed.